libs/execution_context/ebpf_maps.c

```diff
@@ -841,16 +902,34 @@ _create_lru_hash_map(
         goto Exit;
     }
 
+    size_t lru_entry_size;
+    retval = ebpf_safe_size_t_add(EBPF_OFFSET_OF(ebpf_lru_entry_t, key), map_definition->key_size, &lru_entry_size);
+    if (retval != EBPF_SUCCESS) {
+        goto Exit;
+    }
+
     // Align the supplemental value to 8 byte boundary.
     // Pad value_size to next 8 byte boundary and subtract the value_size to get the padding.
-    size_t supplemental_value_size =
-        sizeof(uint64_t) + EBPF_PAD_8(map_definition->value_size) - map_definition->value_size;
+    size_t supplemental_value_size;
+    retval = ebpf_safe_size_t_add(
+        lru_entry_size, EBPF_PAD_8(map_definition->value_size) - map_definition->value_size, &supplemental_value_size);
+    if (retval != EBPF_SUCCESS) {
+        goto Exit;
+    }
 
     retval = _create_hash_map_internal(
         sizeof(ebpf_core_lru_map_t), map_definition, supplemental_value_size, NULL, (ebpf_core_map_t**)&lru_map);
     if (retval != EBPF_SUCCESS)
         goto Exit;
 
+    ebpf_list_initialize(&lru_map->hot_list);
+    ebpf_list_initialize(&lru_map->cold_list);
+    ebpf_lock_create(&lru_map->lock);
+
+    lru_map->current_generation = EBPF_LRU_INITIAL_GENERATION;
+    lru_map->hot_list_size = 0;
+    lru_map->hot_list_limit = min(map_definition->max_entries / EBPF_LRU_GENERATION_COUNT, 1);
```

Comment:
should it be `max`?

Reply:
Hmm, that's a really good point. This is effectively hard coding the limit to 1.

Reply:
We need a better perf test for LRU. The current test just continually evicts the oldest entry, so generation size == 1 performs the same as generation size == max_entries / EBPF_LRU_GENERATION_COUNT.